### PR TITLE
docs: Taking a crack at updating the scripts docs.

### PR DESCRIPTION
**Pull request by seanhealy**
After doing a bit of a dive into the scripts life cycle to setup a git install I thought I'd take a crack at breaking down the life cycle here. I did a really rough start to see what y'all thought before going through each one just to make sure it made sense.

My plan was to break the life cycle hooks for each command out as I have started here.

I was also thinking of explaining some of the configurable commands such as `start` and `test`.

Anyway, hopefully this makes some sort of sense. 😃

### CODE REVIEW

**Comment by zkat** [comment]
cc @neverett for review

File: doc/misc/npm-scripts.md [hunk 1]
```diff
@@ -3,57 +3,90 @@ npm-scripts(7) -- How npm handles the "scripts" field
 
 ## DESCRIPTION
 
-npm supports the "scripts" property of the package.json file, for the
-following scripts:
+The `"scripts"` property of of your `package.json` file supports a number of built in scripts and their preset life cycle events as well as arbitrary scripts which can be executed by running `npm run <stage>`. *Pre* and *post* commands with matching names will be run for those as well (e.g. `premyscript`, `myscript`,
```
Comment (neverett) [inline doc/misc/npm-scripts.md]: s/built in/built-in

File: doc/misc/npm-scripts.md [hunk 2]
```diff
@@ -3,57 +3,90 @@ npm-scripts(7) -- How npm handles the "scripts" field
...
+  * Runs BEFORE the package is packed and published, as well as on local `npm
   install` without any arguments. (See below)
-* prepare:
-  Run both BEFORE the package is packed and published, and on local `npm
-  install` without any arguments (See below). This is run
-  AFTER `prepublish`, but BEFORE `prepublishOnly`.
-* prepublishOnly:
-  Run BEFORE the package is prepared and packed, ONLY on `npm publish`. (See
+* `prepare`:
+  * Run both BEFORE the package is packed and published, and on local `npm install` without any arguments (See below). This is run AFTER `prepublish`, but BEFORE `prepublishOnly`.
```
Comment (neverett) [inline doc/misc/npm-scripts.md]: Should be "Runs", for consistency

File: doc/misc/npm-scripts.md [hunk 3]
```diff
@@ -3,57 +3,90 @@ npm-scripts(7) -- How npm handles the "scripts" field
...
+  * Run by the `npm shrinkwrap` command.
 
-Additionally, arbitrary scripts can be executed by running `npm
-run-script <stage>`. *Pre* and *post* commands with matching
-names will be run for those as well (e.g. `premyscript`, `myscript`,
-`postmyscript`). Scripts from dependencies can be run with `npm explore
-<pkg> -- npm run <stage>`.
+### Life Cycle
+
+npm commands such as `npm install`, `npm publish` will fire life cycle hooks as specified in your `package.json` file. These hooks are as follows for each command.
```
Comment (neverett) [inline doc/misc/npm-scripts.md]: I'd change the comma to "and": "npm commands such as `npm install` and `npm publish`..."

File: doc/misc/npm-scripts.md [hunk 4]
```diff
@@ -3,57 +3,90 @@ npm-scripts(7) -- How npm handles the "scripts" field
...
+
+#### [`npm publish`](../cli/npm-publish.md)
```
Comment (neverett) [inline doc/misc/npm-scripts.md]: I think for this to be linked correctly on the docs site, you'll want to form the link like so: 

```
[`npm publish`](/cli/npm-publish)
```

(This also applies to the links to `npm pack` and `npm install` below.)

File: doc/misc/npm-scripts.md [hunk 5]
```diff
@@ -3,57 +3,90 @@ npm-scripts(7) -- How npm handles the "scripts" field
...
+  - `install`, `postinstall`
+    - Run AFTER the package is installed.
+
+- Also triggers
+  - `prepublish` (when on local)
+  - `prepare` (when on local)
+
+#### `start`
+
+npm run start has an `npm start` shorthand.
```
Comment (neverett) [inline doc/misc/npm-scripts.md]: Suggest wrapping `npm run start` in backticks.

**Review by neverett — COMMENTED** [review]
I left a few small bits of feedback, but otherwise this looks good to me.

**Comment by seanhealy** [comment]
Thanks for the comments! I'll circle back to finish the other sections and make these changes as soon as I can. 😃

**Comment by seanhealy** [comment]
Sorry, this totally fell off my radar but, I'm really stoked that y'all 've picked it back up! 🍻

**Comment by mikemimik** [comment]
Note: will be moving this from [Sprint 2](https://github.com/npm/cli/milestone/3) -> [Sprint 3](https://github.com/npm/cli/milestone/4) and rolled into the next release.

**Comment by darcyclarke** [comment]
Closed out by @mikemimik in https://github.com/npm/cli/pull/729

**Pull request closed without merging**